site/en/workflows.md

```diff
@@ -598,6 +598,13 @@ For detailed descriptions of behaviour and options, see [tesserocr's README](htt
       <td><code>-P find_tables false -P shrink_polygons true</code></td>
       <td>Recommended. Will reuse internal tesseract iterators to produce a complete segmentation with tight polygons instead of bounding boxes where possible</td>
       <td><code>ocrd-tesserocr-segment -I OCR-D-DEWARP-PAGE -O OCR-D-SEG -P find_tables false -P shrink_polygons true</code></td>
+    </tr>
+	<tr data-processor="ocrd-eynollah-segment">
+      <td>ocrd-eynollah-segment</td>
+      <td><code>-P models</code></td>
+      <td>Models can be found <a href="https://qurator-data.de/eynollah/models_eynollah.tar.gz)">here</a> or downloaded with the [OCR-D resource manager](https://ocr-d.de/en/models); <br/>
+      If you didn't download the model with the `resource manager`, for <code>model</code> you need to <strong>pass the local path on your hard drive</strong> as parameter value.</td>
+      <td><code>ocrd-eynollah-segment -I OCR-D-DEWARP-PAGE -O OCR-D-SEG -P models default</code></td>
```

Comment:
```suggestion
      <td><code>ocrd-eynollah-segment -I OCR-D-IMG -O OCR-D-SEG -P models default</code></td>
```

eynollah always uses the "raw" image, i.e. what is in `@imageFilename`. But it's actually a good question, how can we run eynollah on the dewarped page - is that even sensible @vahidrezanezhad?

Reply:
Would need some page-level dewarping processor that runs on the raw image, and provides PAGE files using the dewarped image as `@imageFilename` (because we cannot describe and reconstruct the coordinate transform). Also, because the original file is where we take all meta-data from (including DPI), that processor would need to preserve (re-add) the meta-data to its output image.

We do not have such a processor, yet. Otherwise, I don't see a problem here.

Reply:
> could you add the information on the expected input filegroup to the json description?

The info is kind of there in the form of

```
Default Wiring:
  ['OCR-D-IMG', 'OCR-D-SEG-PAGE', 'OCR-D-GT-SEG-PAGE'] -> ['OCR-D-SEG-LINE']
```

But I get what you mean, we should probably explain that it doesn't matter much what input file group you specify (unless dewarping is involved as @berstky explained), because it will always use the `@imageFilename` which is the same for every PAGE-XML in every filegroup.

Reply:
ah, didn't notice this even when looking for it 🙈 

not really sure how to solve this yet. but as a user who goes down along the steps in the workflow guide I would be quite annoyed if I do several preprocessing steps and then get to know that it was actually useless for this processor. I would need this info right at the beginning so I can skip the irrelevant steps and jump right down to the segmentation

Reply:
> I would be quite annoyed if I do several preprocessing steps and then get to know that it was actually useless for this processor.

It's true that it gets confusing if a processor does as many (still unseparable) steps in one go, so we should give example workflows with eynollah that make it clear what steps it already contains. However, you still need e.g. a good binarization for the OCR, so it's not like you could replace all steps until OCR with eynollah necessarily.

The reason why we use the raw image for eynollah and not the `workspace.image_from_page` mechanism as in most other processors, is purely technical, because we'd lose the image metadata that eynollah needs to calculate scale. Hopefully we can get rid of this limitation soon, so that you could use e.g. the despeckled page image if you want to (not saying that this would make sense per se, just that I would like to give users the opportunity).

Reply:
> eynollah always uses the "raw" image, i.e. what is in `@imageFilename`. But it's actually a good question, how can we run eynollah on the dewarped page - is that even sensible @vahidrezanezhad?

Actually, the input can be any form of the raw image. But for most documents, the best performance is achieved with raw input. By the way, if the raw input is strongly dark or bright (flashed ) the binary would yield a better result. However for those documents, you don't need the binarized image you can just turn -ib (-input_binary) and eynollah would do binarization at the first stage.

Reply:
@vahidrezanezhad, I am surprised to find Otsu in here. It looks like you need binarized input for the region models only – is that correct?

In that case, @kba we should indeed strive for a better OCR-D integration ASAP by querying for _both_ the raw _and_ the binarized image (so workflows can do better binarization externally, including sbb-binarize).

Reply:
so for the moment: we could add a note saying that `ocrd-eynollah-segment` works well with less preprocessing steps and refer to some example workflow with enollay which we add at the end of the workflow guide

Reply:
> requires good documentation of each specific processor's requirements so one can configure sensible workflows.

@kba in that light we should at least provide some basic docstring in eynollah's `process()` which then shows up in `--help`.